pkg/apis/mayadata.io/v1alpha1/backup_schedule.go

```diff
@@ -36,13 +36,31 @@ type DMaaSBackup struct {
 	Status DMaaSBackupStatus `json:"status"`
 }
 
+// DMaasBackupState represents the state of DMaasBackup
+type DMaasBackupState string
+
+const (
+	// DMaaSBackupStateEmpty means dmaasbackup is active
+	DMaaSBackupStateEmpty DMaasBackupState = ""
+
+	// DMaaSBackupStateActive means dmaasbackup is active
+	DMaaSBackupStateActive DMaasBackupState = "Active"
+
+	// DMaaSBackupStatePaused means dmaasbackup is paused
+	DMaaSBackupStatePaused DMaasBackupState = "Paused"
+)
+
 // DMaaSBackupSpec defines the spec for DMaaSBackup resource
 type DMaaSBackupSpec struct {
 	// State defines if given DMaaSBackup is active or not
 	// Default value is Active
 	// +optional
+	State DMaasBackupState `json:"state,omitempty"`
```

Comment:
is this supposed to be pointer?

Reply:
I think keeping it pointer may not help here since State will always have value.. User may not provide it but it will be updated to Active.

Reply:
setting spec related fields in controller is not a good idea.. probably webhook can do that..
its good to even avoid setting in webhook as users might have stored the yamls in github or some place for later use, and so its better to keep consistent.
Reason for pointer is to know that user didn't really provided "empty" but it just happened to be empty

Reply:
empty/active both are considered as active. So why should we worry if user has set it  to "empty" or happened to be "empty". We can keep it pointer if we are checking if value is there or not. Here we also need to check if value is "active" or "paused".

Reply:
information that user really provided "empty" or not provided anything can be really useful if value of the default is getting changed later on.
You remember the field 'thinprovisioning'. If its empty, its called as enabled. But, later we actually want to disable that feature and we had tough time to know whether user really left it as empty or he made it as empty. Then, we introduced new field 'thickprovisioning'.
Point I'm trying to say is: that is an info and good to have it.

Reply:
This can be taken to next PR if we both agree it is fine.